src/Data/Slug.purs

```diff
@@ -0,0 +1,120 @@
+module Data.Slug
+  ( Slug
+  , generate
+  , parse
+  , toString
+  , truncate
+  ) where
+
+import Prelude
+
+import Data.Argonaut.Decode (class DecodeJson, JsonDecodeError(..), decodeJson)
+import Data.Argonaut.Encode (class EncodeJson, encodeJson)
+import Data.Array as Array
+import Data.CodePoint.Unicode (isLatin1)
+import Data.Either (note)
+import Data.Maybe (Maybe(..))
+import Data.String as String
+import Data.String.CodePoints (codePointFromChar, fromCodePointArray, toCodePointArray)
+import Data.String.Pattern (Pattern(..), Replacement(..))
+
+-- | A `Slug` represents a string value which is guaranteed to have the
+-- | following qualities:
+-- |
+-- | - it is not empty
+-- | - it consists of groups of characters separated by `-` dashes,
+-- |   where the slug cannot begin or end with a dash, and there can
+-- |   never be two dashes in a row.
+-- |
+-- | Example: `Slug "this-is-an-article-slug"`
+newtype Slug = Slug String
```

Comment:
This was copied from https://github.com/thomashoneyman/purescript-slug/blob/main/src/Slug.purs, removing the following guarantees: "every character with a defined notion of case is lower-cased" and "it consists of _alphanumeric_ groups of characters".